eldritch/expansions/seaside/abilities.py

```diff
@@ -0,0 +1,122 @@
+import typing
+
+from eldritch import assets
+from eldritch import characters
+from eldritch import events
+from eldritch import values
+
+if typing.TYPE_CHECKING:
+  from eldritch.eldritch import GameState
+
+
+class AbnormalFocus(assets.Asset):
+  def __init__(self):
+    super().__init__("Abnormal Focus")
+
+  def get_interrupt(self, event, owner: characters.BaseCharacter, state):
+    if isinstance(event, events.Upkeep) and event.character == owner:
+      return events.MoveSliders(owner, {slider + "_slider": 0 for slider in owner.sliders()})
+    return None
+
+
+class BreakingTheLimits(assets.Asset):
+  def __init__(self):
+    super().__init__("Breaking the Limits")
+    self.tokens["sanity"] = 0
+    self.tokens["stamina"] = 0
+
+  def get_interrupt(self, event, owner, state):
+    if isinstance(event, events.Upkeep) and event.character == owner:
+      return events.Sequence([
+          events.RemoveToken(self, "sanity", owner, self.tokens["sanity"]),
+          events.RemoveToken(self, "stamina", owner, self.tokens["stamina"]),
+      ], owner)
+    return None
+
+  def get_usable_interrupt(self, event, owner, state):
+    if (
+        not isinstance(event, events.SliderInput)
+        or event.character != owner
+        or sum(self.tokens.values()) >= 3
+        or (owner.stamina == 1 and owner.sanity == 1)
+    ):
+      return None
+    spend = events.SpendChoice(
+        owner, prompt="Break the limits?",
+        choices=["No", "Yes"],
+        spends=[
+            None,
+            values.FlexibleRangeSpendPrerequisite(
+                ["sanity", "stamina"], 1, 3-sum(self.tokens.values())
+            )
+        ]
+    )
+
+    return events.Sequence([
+        spend,
+        events.AddTokenMap(self, values.Calculation(spend, "spend_map"), owner)
+    ], owner)
+
+  def get_bonus(self, check_type, attributes, owner, state):
+    if check_type == "abnormal_focus":
+      return sum(self.tokens.values())
+    return 0
+
+
+class Synergy(assets.Asset):
+  def __init__(self):
+    super().__init__("Synergy")
+
+  def get_bonus(
+      self, check_type, attributes, owner: characters.BaseCharacter, state: "GameState"
+  ):
+    # TODO: Call Stack enforcement to guard against infinite loops.
+    n_allies = len([ally for ally in owner.possessions if ally.deck == "allies"])
```

Comment:
You'll want to use `getattr(ally, "deck", "")` since not every asset has a `deck` attribute.